node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/MetricsResponse.java

```diff
@@ -56,104 +57,120 @@ private MetricsResponse(Slime response, NodeList applicationNodes, NodeRepositor
+    private boolean clusterIsStable(Node node, NodeList applicationNodes, NodeRepository nodeRepository) {
+        ClusterSpec cluster = node.allocation().get().membership().cluster();
+        return Autoscaler.stable(applicationNodes.cluster(cluster.id()), nodeRepository);
+    }
+
     public static MetricsResponse empty() { return new MetricsResponse(List.of()); }
 
     /** The metrics this can read */
     private enum Metric {
 
         cpu { // a node resource
-            public String metricResponseName() { return "cpu.util"; }
-            double convertValue(double metricValue) { return (float)metricValue / 100; } // % to ratio
+            public List<String> metricResponseNames() { return List.of("cpu.util"); }
+            double computeFinal(List<Double> values) {
+                return values.stream().mapToDouble(v -> v).average().orElse(0) / 100; // % to ratio
+            }
         },
         memory { // a node resource
-            public String metricResponseName() { return "mem.util"; }
-            double convertValue(double metricValue) { return (float)metricValue / 100; } // % to ratio
+            public List<String> metricResponseNames() { return List.of("mem.util"); }
+            double computeFinal(List<Double> values) {
+                return values.stream().mapToDouble(v -> v).average().orElse(0) / 100; // % to ratio
+            }
         },
         disk { // a node resource
-            public String metricResponseName() { return "disk.util"; }
-            double convertValue(double metricValue) { return (float)metricValue / 100; } // % to ratio
+            public List<String> metricResponseNames() { return List.of("disk.util"); }
+            double computeFinal(List<Double> values) {
+                return values.stream().mapToDouble(v -> v).average().orElse(0) / 100; // % to ratio
+            }
         },
         generation { // application config generation active on the node
-            public String metricResponseName() { return "application_generation"; }
-            double convertValue(double metricValue) { return (float)metricValue; } // Really a long
-            double defaultValue() { return -1.0; }
+            public List<String> metricResponseNames() { return List.of("application_generation"); }
+            double computeFinal(List<Double> values) {
+                return values.stream().mapToDouble(v -> v).min().orElse(-1);
+            }
         },
         inService {
-            public String metricResponseName() { return "in_service"; }
-            double convertValue(double metricValue) { return (float)metricValue; } // Really a boolean
-            double defaultValue() { return 1.0; }
+            public List<String> metricResponseNames() { return List.of("in_service"); }
+            double computeFinal(List<Double> values) {
+                // Really a boolean. Default true. If any is oos -> oos.
+                return values.stream().anyMatch(v -> v == 0) ? 0 : 1;
```

Comment:
Should we return 0 if `values` is empty?

Reply:
Not all services send this (yet) so it needs to default to 1.